### PR TITLE
[price-pusher] add injective price listener

### PR DESCRIPTION
**Pull request by 0xfirefist**

### CODE REVIEW

File: price_pusher/src/index.ts
Comment (ali-bahjati) [inline price_pusher/src/index.ts:18]: Can you also update the readme with this new configs and values? You would need to update the compose files too.

File: price_pusher/package.json
Comment (ali-bahjati) [inline price_pusher/package.json:43]: I think we need to bump version as major for this change.

File: price_pusher/src/index.ts
Comment (ali-bahjati) [inline price_pusher/src/index.ts:90]: minor: I think `injective` and `evm` can have the same run function (or use a common function for running controller, ...) as many things are similar in your code.

File: price_pusher/src/injective.ts
Comment (ali-bahjati) [inline price_pusher/src/injective.ts:50]: What happens if a priceId does not exist?

File: price_pusher/src/injective.ts
Comment (ali-bahjati) [inline price_pusher/src/injective.ts:20]: I see that this is very similar to EVM. Maybe it's good to extract common identical methods to `PriceListener`.

**Review by ali-bahjati — COMMENTED** [review]
This is awesome. I really appreciate that you are improving the quality of this code. I left some comments and will approve once you address them.

File: price_pusher/src/index.ts
Comment (0xfirefist) [reply price_pusher/src/index.ts:18]: I am not sure if this file is final yet. 
I have added the code to it just so that injective pieces are working fine. 
I will refactor this file too and then will update the README

File: price_pusher/package.json
Comment (0xfirefist) [reply price_pusher/package.json:43]: will send a pr with readme and version changes. which will be the final one after refactoring.

File: price_pusher/src/index.ts
Comment (0xfirefist) [reply price_pusher/src/index.ts:90]: as mentioned above
will clean this file in separate PR

File: price_pusher/src/index.ts
Comment (jayantk) [reply price_pusher/src/index.ts:18]: command line options will be annoying as the configuration becomes more complicated. I suggest moving this stuff to a config file. I note that there's already a price config file, so maybe it goes in there.

File: price_pusher/src/injective.ts
Comment (jayantk) [reply price_pusher/src/injective.ts:20]: agree with this. i think the latestPriceInfo / update / get methods could be moved up to priceListener

I wouldn't move the pollPrices method even though it is very similar to the evm method, as you want to give implementations the flexibility to implement the reading logic in different ways.

File: price_pusher/src/index.ts
Comment (jayantk) [reply price_pusher/src/index.ts:90]: FYI if you want to do this stuff in a separate PR, i suggest leaving a code comment like 

`// FIXME: unify with evmRun() below once the interfaces are figured out.`

then people will understand what you're trying to do when reading the PR and won't leave comments about it

**Review by jayantk — COMMENTED** [review]
i think extracting common methods like @ali-bahjati says is a good thing to do. If ali is happy, then I am; no need to wait for me to approve.

**Review by ali-bahjati — APPROVED** [review]

**Pull request merged**
